src/panels/lovelace/editor/unused-entities/hui-unused-entities.ts

```diff
@@ -0,0 +1,211 @@
+    if (!this._config || !this._hass || !this.lovelace) {
+      return;
+    }
+
+    if (this.lovelace.editMode === false) {
+      return;
+    }
+
+    return html`
+      ${this.renderStyle()}
+      <ha-card header="Unused entities">
+        <div class="card-content">
+          These are the entities that you have available, but are not in your
+          Lovelace UI yet.
+          <br />Select the entities you want to add to a card and then click the
+          add card button.
+        </div>
+        <table
+          class="entities"
+          @entity-selection-changed=${this._handleSelectionChanged}
+        >
+          <thead>
+            <tr>
+              <th>Entity</th>
+              <th>Entity id</th>
+              <th>Domain</th>
+              <th>Last changed</th>
+            </tr>
+          </thead>
+          <tbody>
+            ${this._unusedEntities.map((entity) => {
+              return html`
+                <hui-select-row
+                  .hass=${this._hass}
+                  .entity=${entity}
+                ></hui-select-row>
+              `;
+            })}
+          </tbody>
+        </table>
+      </ha-card>
+      <paper-fab
+        elevated="2"
+        class="${classMap({
+          rtl: computeRTL(this._hass),
+        })}"
+        icon="hass:plus"
+        title="${this._hass.localize("ui.panel.lovelace.editor.edit_card.add")}"
+        @click="${this._selectView}"
+      ></paper-fab>
+    `;
+  }
+
+  private renderStyle(): TemplateResult {
+    return html`
+      <style>
+        :host {
+          background: var(--lovelace-background);
+          padding: 16px;
+        }
+        paper-fab {
+          position: sticky;
+          float: right;
+          bottom: 16px;
+          z-index: 1;
+        }
+        paper-fab.rtl {
+          float: left;
+        }
+
+        .entities {
+          border-collapse: collapse;
+          width: 100%;
+          margin: auto;
+          table-layout: fixed;
+        }
+
+        .entities th {
+          text-align: left;
+          padding: 12px 24px;
+          vertical-align: middle;
+          font-size: 13px;
+          line-height: 17px;
+        }
+
+        .entities hui-select-row {
+          border-bottom: 1px solid #e0e0e0;
+        }
+
+        .entities tbody hui-select-row:hover {
+          background-color: var(--table-row-alternative-background-color, #eee);
+        }
+
+        .entities td {
+          padding: 4px;
+        }
+
+        .entities td:nth-child(3) {
+          white-space: pre-wrap;
+          word-break: break-word;
+        }
+      </style>
+    `;
+  }
+
+  private _getUnusedEntities(): void {
+    if (!this._hass) {
+      return;
+    }
+    this._selectedEntities = [];
+    this._unusedEntities = computeUnusedEntities(this._hass, this._config!);
+  }
+
+  private _handleSelectionChanged(ev: any): void {
+    if (ev.detail.selected) {
+      this._selectedEntities.push(ev.detail.entity);
```

Comment:
This will not cause a re-render, unselecting does

Reply:
`this_selectedEntities` isn't a property so both won't re-render?

Reply:
Check. 

Just as as note for when we're passing data around: Lit works by doing identity checks `===`, and so will detect that an object is different if we copied it, but not if we pushed data to it. So if you were to have this be a property and pass it around, `[...this._selectedEnttiies, ev.detail.entity]` would be the way to make a copy with the change.